pkg/controller/eks/nodegroup/controller.go

```diff
@@ -219,3 +229,70 @@ func (t *tagger) Initialize(ctx context.Context, mg resource.Managed) error {
 	}
 	return errors.Wrap(t.kube.Update(ctx, cr), errKubeUpdateFailed)
 }
+
+func versionHasChanged(cr *manualv1alpha1.NodeGroup, ng *awseks.DescribeNodegroupOutput) bool {
+	if ng.Nodegroup.LaunchTemplate != nil && cr.Spec.ForProvider.LaunchTemplate != nil {
+		if !reflect.DeepEqual(ng.Nodegroup.LaunchTemplate.Version, cr.Spec.ForProvider.LaunchTemplate.Version) {
+			return true
+		}
+	}
+	return !reflect.DeepEqual(ng.Nodegroup.Version, cr.Spec.ForProvider.Version)
+}
+
+func generateInput(ctx context.Context, e *external, cr *manualv1alpha1.NodeGroup, ng *awseks.DescribeNodegroupOutput) (*awseks.UpdateNodegroupVersionInput, error) {
+	input := &awseks.UpdateNodegroupVersionInput{
+		ClusterName:   &cr.Spec.ForProvider.ClusterName,
+		NodegroupName: awsclient.String(meta.GetExternalName(cr)),
+	}
+
+	var customAMI bool
+	if ng.Nodegroup.LaunchTemplate != nil {
+		crVersion, err := strconv.ParseInt(*cr.Spec.ForProvider.LaunchTemplate.Version, 10, 64)
+		if err != nil {
+			return &awseks.UpdateNodegroupVersionInput{}, awsclient.Wrap(err, "cannot convert template version")
+		}
+		describeInput := &awsec2.DescribeLaunchTemplateVersionsInput{
+			LaunchTemplateName: cr.Spec.ForProvider.LaunchTemplate.Name,
+			Versions:           []string{*cr.Spec.ForProvider.LaunchTemplate.Version},
+		}
+		ltp, err := e.ec2Client.DescribeLaunchTemplateVersions(ctx, describeInput)
+		if err != nil {
+			return &awseks.UpdateNodegroupVersionInput{}, awsclient.Wrap(err, "cannot describe launchtemplate version")
+		}
+		for _, v := range ltp.LaunchTemplateVersions {
+			if *v.VersionNumber == crVersion {
+				if v.LaunchTemplateData.ImageId != nil {
+					customAMI = true
+				}
+				break
+			}
+		}
+		input.LaunchTemplate = &types.LaunchTemplateSpecification{
+			Id:      cr.Spec.ForProvider.LaunchTemplate.ID,
+			Name:    cr.Spec.ForProvider.LaunchTemplate.Name,
+			Version: cr.Spec.ForProvider.LaunchTemplate.Version,
+		}
+	}
+
+	if !customAMI {
+		input.Version = cr.Spec.ForProvider.Version
+	}
```

Comment:
This part can cause unexpected behaviour when the `releaseVersion` used by the NodeGroup is not the latest version.

According to the API Documentation, passing the same version as is currently used in the `UpdateNodeGroupVersionInput` will cause an update to the latest `releaseVersion` of the current Kubernetes version.

I've tested it with a NodeGroup at `1.23.7-20220802` and it behaved as expected
```
baldorch$ aws eks update-nodegroup-version \
  --cluster-name "$cluster" \
  --nodegroup-name "$nodegroup" \
  --launch-template name="$launchtemplate",version=3 \
  --kubernetes-version 1.23
  
{
    "update": {
        "id": "123450",
        "status": "InProgress",
        "type": "VersionUpdate",
        "params": [
            {
                "type": "Version",
                "value": "1.23"
            },
            {
                "type": "ReleaseVersion",
                "value": "1.23.9-20220824"
            },
            {
                "type": "LaunchTemplateName",
                "value": "my-launch-template-name"
            },
            {
                "type": "LaunchTemplateVersion",
                "value": "3"
            }
        ],
        "createdAt": "2022-09-07T15:20:49.903000+02:00",
        "errors": []
    }
}
```

When I omit the `--kubernetes-version` it only updates the LaunchTemplate:
```
baldorch$ aws eks update-nodegroup-version \
  --cluster-name "$cluster" \
  --nodegroup-name "$nodegroup" \
  --launch-template name="$launchtemplate",version=3

{
    "update": {
        "id": "123450",
        "status": "InProgress",
        "type": "VersionUpdate",
        "params": [
            {
                "type": "LaunchTemplateName",
                "value": "my-launch-template-name"
            },
            {
                "type": "LaunchTemplateVersion",
                "value": "3"
            }
        ],
        "createdAt": "2022-09-07T15:39:59.233000+02:00",
        "errors": []
    }
}
```

The version should not be added to the `UpdateNodegroupVersionInput` if it did not change. And if it changed and it gets added it still ignores the `releaseVersion` if it got passed. So instead of updating to a specific AMI it will update to the latest.